### PR TITLE
Revise intro to address question of realistic models

### PR DESCRIPTION
**Pull request by cboettig**
@mlap @milliechapman  @karinorman  okay, tried to  tackle the editor's concerns about why we don't use more realistic simulations.  Maybe I came over a little too strong here; edits appreciated.

### CODE REVIEW

File: manuscript/manuscript.Rmd [hunk 1]
```diff
@@ -23,18 +23,17 @@ authors:
...
-  fisheries quotas and in managing ecological tipping points.
-  Four appendices with annotated code provide a tangible introduction to researchers 
-  looking to adopt, evaluate, or extend these approaches.
+  In this paper, we illustrate and discuss the potential of a promising corner of machine learning known as _reinforcement learning_ (RL) to help tackle the most challenging conservation decision problems.
+  RL is uniquely well suited to conservation and global change challenges for three reasons: 
+  (1) RL explicitly focuses on designing an agent who _interacts_ with an environment which is dynamic and uncertain,
+  (2) RL approaches do not require massive amounts of data, 
+  (3) RL approaches would utilize rather than replace existing models, simulations, and the knowledge they contain.
+  While this potential may be compelling, the challenges involved in successfully deploying RL-based management to realistic scenarios are formidable --
+  the required expertise and computational cost may place these applications may be beyond the reach of all but international technology firms.
```
Comment (milliechapman) [inline manuscript/manuscript.Rmd:32]: "the required expertise and computational cost may place these applications beyond the reach of all but international technology firms."

File: manuscript/manuscript.Rmd [hunk 2]
```diff
@@ -23,18 +23,17 @@ authors:
...
+  Such an advent would represent a sea-change not only in what problems we can solve, but what stakeholders are involved in solving them.
```
Comment (milliechapman) [inline manuscript/manuscript.Rmd:33]: this feels a vague --we might consider removing the stakeholder part of this sentence. doesnt seem super necessary in the abstract..

File: manuscript/manuscript.Rmd [hunk 3]
```diff
@@ -112,11 +111,36 @@ Rich, processes-based simulations such as the SORTIE model in forest management
 Decision-theoretic approaches based on optimal control techniques can only find the best strategy in the simplest of ecological models; the so called "curse of dimensionality" makes problems with a large number of states or actions intractable by conventional methods [@Possingham2006; @Marescot2013; @ferrer-mestres_k-n-momdps_2021; @chades_primer_2021].
 Neural-network-based RL techniques, referred to as *deep RL*, have proven particularly effective in problems involving complex, high-dimensional spaces that have previously proven intractable to classical methods.
 
+While deep RL may have the potential to open up such intractable problems, it also risks making those problems tractable only by stakeholders with access to extensive computational resources and expertise.
+It is notable that the landmark advances cited above have been solved not by academic teams but by specialized research teams of international technology firms such as Alphabet.
```
Comment (milliechapman) [inline manuscript/manuscript.Rmd:115]: 💯 , 🔥

File: manuscript/manuscript.Rmd [hunk 4]
```diff
@@ -112,11 +111,36 @@ Rich, processes-based simulations such as the SORTIE model in forest management
...
+Our implementations of these models in two python modules published to `PyPi` code archive include many such variations which represent open problems for RL.
 We include extensive appendices with carefully annotated code which should allow readers to both reproduce and extend this analysis.
 
+We do recognize the need for more complex, "precise" models to derive quanitatively precise management policy in specific management problems such as fisheries, forestry, or climate change mitigation efforts. 
+The examples here should provide ecologists with the greater background to engage in the collaborative development of such methods, even as they also highlight challenges which ecological problems pose to existing techniques.
+Even the simple models we consider here contain many sources of complexity -- significant stochasticity, non-linear dynamics, hysteresis -- that are likely commonplace to ecological problems but have not been well represtented in RL research to date. 
+Such work is beyond the scope of any one paper: it will require not only more computational resources, but more collaborations between large teams of ecologists and computer scientists.
```
Comment (milliechapman) [inline manuscript/manuscript.Rmd:142]: this sentence seems like it should go after the first sentence in the paragraph?

File: manuscript/manuscript.Rmd [hunk 5]
```diff
@@ -112,11 +111,36 @@ Rich, processes-based simulations such as the SORTIE model in forest management
...
+These models provide an ideal first benchmark for evaluating the performance of emerging methods of deep RL for several reasons:
+Firstly, for some cases the optimal solution is already known, providing a clear standard-of-comparison to evaluate RL performance. 
+Prior work sometimes overlooks this essential step, assuming that whatever behavior an RL agent produces is sufficiently optimal or otherwise an accurate proxy for behavior arising from animal learning or evolution. 
+As our evaluations will illustrate, such an assumption can be quickly misleading. 
+Second, these models are already widely studied and will be familiar to many readers: @Schaefer1954 is a staple of fisheries management textbooks and practice, with over 2800 citations, while @May1977 has become a canonical model of threshholds and tipping points which still continues to dominate how many ecologists think about these phenomena [@Scheffer2009].
```
Comment (milliechapman) [inline manuscript/manuscript.Rmd:132]: I think you could remove everything after the colon and go strait to the next sentence.

File: manuscript/manuscript.Rmd [hunk 6]
```diff
@@ -112,11 +111,36 @@ Rich, processes-based simulations such as the SORTIE model in forest management
...
+As Richard Levins successfully established in his classic paper on the principles of model building [@Levins1966], model complexity must not be mistaken for model realism.
+Levins esposed simple mechanistic models which satisfy the goals of being both _realistic_ and _general_. 
+More complex models such as those used in fisheries to guide the management of specific stocks typically sacrifice _generality_ for _precision_, a distinction Bob May characterized as "tactical models" (precise) rather than "strategic models" (general). 
+Such simple, realistic and general models are still the bedrock of most theory and practice today, even when implementation turns to tactical models for quantitative guidance (for instance, the notion of maximum sustainable yield, MSY, in fisheries, or $R_0$ in epidemiology, remain important concepts in management even though they do not exist in typical tactical models).
+These models provide an ideal first benchmark for evaluating the performance of emerging methods of deep RL for several reasons:
```
Comment (milliechapman) [inline manuscript/manuscript.Rmd:128]: I like this added paragraph -- I think it provides a succinct justification for the use of simple models throughout the paper!

**Review by milliechapman — COMMENTED** [review]
really like the edits -- made some minor comments below!

File: manuscript/manuscript.Rmd [hunk 7]
```diff
@@ -23,18 +23,17 @@ authors:
...
+  RL is uniquely well suited to conservation and global change challenges for three reasons: 
+  (1) RL explicitly focuses on designing an agent who _interacts_ with an environment which is dynamic and uncertain,
+  (2) RL approaches do not require massive amounts of data, 
+  (3) RL approaches would utilize rather than replace existing models, simulations, and the knowledge they contain.
+  While this potential may be compelling, the challenges involved in successfully deploying RL-based management to realistic scenarios are formidable --
+  the required expertise and computational cost may place these applications may be beyond the reach of all but international technology firms.
+  Such an advent would represent a sea-change not only in what problems we can solve, but what stakeholders are involved in solving them.
+  Ecologists must establish a better understanding of how these algorithms work and fail if we are to realize this potential and avoid the pitfalls such a transition would bring.
```
Comment (mlap) [inline manuscript/manuscript.Rmd:34]:
```suggestion
  Ecologists must establish a better understanding of these algorithms if we are to realize the potential of deep RL and avoid the pitfalls such a transition would bring.
```
I'm not set on my suggestion here but I think this sentence could be clearer.

File: manuscript/manuscript.Rmd [hunk 8]
```diff
@@ -112,11 +111,36 @@ Rich, processes-based simulations such as the SORTIE model in forest management
...
 Neural-network-based RL techniques, referred to as *deep RL*, have proven particularly effective in problems involving complex, high-dimensional spaces that have previously proven intractable to classical methods.
 
+While deep RL may have the potential to open up such intractable problems, it also risks making those problems tractable only by stakeholders with access to extensive computational resources and expertise.
+It is notable that the landmark advances cited above have been solved not by academic teams but by specialized research teams of international technology firms such as Alphabet.
+Precise estimates of computational resources used in that research are difficult to establish, but previous estimates benchmarked against commercially avilable cloud computing platforms place the training of a single model at over $30 million [@alphacost],
```
Comment (mlap) [inline manuscript/manuscript.Rmd:116]:
```suggestion
Precise estimates of computational resources used in that research are difficult to establish, but previous estimates benchmarked against commercially available cloud computing platforms place the training of a single model at over $30 million [@alphacost],
```

File: manuscript/manuscript.Rmd [hunk 9]
```diff
@@ -112,11 +111,36 @@ Rich, processes-based simulations such as the SORTIE model in forest management
...
+These models provide an ideal first benchmark for evaluating the performance of emerging methods of deep RL for several reasons:
+Firstly, for some cases the optimal solution is already known, providing a clear standard-of-comparison to evaluate RL performance. 
+Prior work sometimes overlooks this essential step, assuming that whatever behavior an RL agent produces is sufficiently optimal or otherwise an accurate proxy for behavior arising from animal learning or evolution. 
+As our evaluations will illustrate, such an assumption can be quickly misleading. 
+Second, these models are already widely studied and will be familiar to many readers: @Schaefer1954 is a staple of fisheries management textbooks and practice, with over 2800 citations, while @May1977 has become a canonical model of threshholds and tipping points which still continues to dominate how many ecologists think about these phenomena [@Scheffer2009].  
+Many readers can thus benefit from exisiting knowledge and intutition about the behavior and implications of these models in interpreting the performance of deep RL, something that would not be possible with a more complex model.
```
Comment (mlap) [inline manuscript/manuscript.Rmd:133]:
```suggestion
Many readers can thus benefit from existing knowledge and intuition about the behavior and implications of these models in interpreting the performance of deep RL, something that would not be possible with a more complex model. 
```

File: manuscript/manuscript.Rmd [hunk 10]
```diff
@@ -112,11 +111,36 @@ Rich, processes-based simulations such as the SORTIE model in forest management
...
+Firstly, for some cases the optimal solution is already known, providing a clear standard-of-comparison to evaluate RL performance. 
+Prior work sometimes overlooks this essential step, assuming that whatever behavior an RL agent produces is sufficiently optimal or otherwise an accurate proxy for behavior arising from animal learning or evolution. 
+As our evaluations will illustrate, such an assumption can be quickly misleading. 
+Second, these models are already widely studied and will be familiar to many readers: @Schaefer1954 is a staple of fisheries management textbooks and practice, with over 2800 citations, while @May1977 has become a canonical model of threshholds and tipping points which still continues to dominate how many ecologists think about these phenomena [@Scheffer2009].  
+Many readers can thus benefit from exisiting knowledge and intutition about the behavior and implications of these models in interpreting the performance of deep RL, something that would not be possible with a more complex model. 
+It would be unwise to consider applications of RL on more complex bespoke models without first analyzing these classic and well-understood cases.
```
Comment (mlap) [inline manuscript/manuscript.Rmd:134]: I am anticipating kickback at this sentence. I'm battling with whether we should spell out why it would be unwise or let the reader fill this in using some of the previous points made.

File: manuscript/manuscript.Rmd [hunk 11]
```diff
@@ -112,11 +111,36 @@ Rich, processes-based simulations such as the SORTIE model in forest management
...
+Third, these models include or can easily be extended to contexts for which the optimal management policy is unknown or inaccessible to classical methods.
+Our implementations of these models in two python modules published to `PyPi` code archive include many such variations which represent open problems for RL.
```
Comment (mlap) [inline manuscript/manuscript.Rmd:136]:
```suggestion
Our implementations of these models have been published to the python-based `PyPi` code archive and include many such variations which represent open problems for RL.
```

File: manuscript/manuscript.Rmd [hunk 12]
```diff
@@ -112,11 +111,36 @@ Rich, processes-based simulations such as the SORTIE model in forest management
...
 We include extensive appendices with carefully annotated code which should allow readers to both reproduce and extend this analysis.
 
+We do recognize the need for more complex, "precise" models to derive quanitatively precise management policy in specific management problems such as fisheries, forestry, or climate change mitigation efforts. 
+The examples here should provide ecologists with the greater background to engage in the collaborative development of such methods, even as they also highlight challenges which ecological problems pose to existing techniques.
```
Comment (mlap) [inline manuscript/manuscript.Rmd:140]:
```suggestion
The examples here should provide ecologists with a greater background to engage in the collaborative development of such methods, while also highlighting challenges which ecological problems pose to existing techniques.
```

**Review by mlap — COMMENTED** [review]
Initial pass at the added parts, I'll do another read through in a bit. Looks good!

**Comment by karinorman** [comment]
Hey - we caught COVID and will be without childcare for a few more days, so may take me a bit to get to this!

File: manuscript/manuscript.Rmd [hunk 13]
```diff
@@ -112,11 +111,36 @@ Rich, processes-based simulations such as the SORTIE model in forest management
...
-We demonstrate that RL-based approaches are capable but by no means a magic bullet: reasonable solutions require careful design of training environments, choice of RL algorithms, tuning and evaluation, as well as substantial computational power.
-Our examples are intentionally simple, aiming to provide a clear template for understanding that could be easily extended to cover more realistic conditions. 
+We focus on examining the potential and limitations of deep RL through the lens of simple, classical models.
+Over a century of theory and practice in ecology has demonstrated that simple models can provide meaningful insights which improve management outcomes [@Getz2017].
+As Richard Levins successfully established in his classic paper on the principles of model building [@Levins1966], model complexity must not be mistaken for model realism.
+Levins esposed simple mechanistic models which satisfy the goals of being both _realistic_ and _general_.
```
Comment (mlap) [inline manuscript/manuscript.Rmd:125]: I think we could make do without this sentence.

File: manuscript/manuscript.Rmd [hunk 14]
```diff
@@ -112,11 +111,36 @@ Rich, processes-based simulations such as the SORTIE model in forest management
...
+More complex models such as those used in fisheries to guide the management of specific stocks typically sacrifice _generality_ for _precision_, a distinction Bob May characterized as "tactical models" (precise) rather than "strategic models" (general). 
+Such simple, realistic and general models are still the bedrock of most theory and practice today, even when implementation turns to tactical models for quantitative guidance (for instance, the notion of maximum sustainable yield, MSY, in fisheries, or $R_0$ in epidemiology, remain important concepts in management even though they do not exist in typical tactical models).
+These models provide an ideal first benchmark for evaluating the performance of emerging methods of deep RL for several reasons:
+Firstly, for some cases the optimal solution is already known, providing a clear standard-of-comparison to evaluate RL performance. 
+Prior work sometimes overlooks this essential step, assuming that whatever behavior an RL agent produces is sufficiently optimal or otherwise an accurate proxy for behavior arising from animal learning or evolution.
```
Comment (mlap) [inline manuscript/manuscript.Rmd:130]: Seems like we should cite this sentence.

File: manuscript/manuscript.Rmd [hunk 15]
```diff
@@ -112,11 +111,36 @@ Rich, processes-based simulations such as the SORTIE model in forest management
...
+Third, these models include or can easily be extended to contexts for which the optimal management policy is unknown or inaccessible to classical methods.
+Our implementations of these models in two python modules published to `PyPi` code archive include many such variations which represent open problems for RL.
 We include extensive appendices with carefully annotated code which should allow readers to both reproduce and extend this analysis.
 
+We do recognize the need for more complex, "precise" models to derive quanitatively precise management policy in specific management problems such as fisheries, forestry, or climate change mitigation efforts. 
+The examples here should provide ecologists with the greater background to engage in the collaborative development of such methods, even as they also highlight challenges which ecological problems pose to existing techniques.
+Even the simple models we consider here contain many sources of complexity -- significant stochasticity, non-linear dynamics, hysteresis -- that are likely commonplace to ecological problems but have not been well represtented in RL research to date.
```
Comment (mlap) [inline manuscript/manuscript.Rmd:141]: I'd consider stochastic and non-linear dynamics to be common in RL, hysteretic dynamics less common.

Comment (cboettig) [reply manuscript/manuscript.Rmd:141]: good call, we should re-word that. (though I did try to hedge with 'significant', and of course hysteresis is just one of the many fun things non-linearity can do :-) ). 
Do you think the sentence would read okay if we merely omitted the examples between the `--`?

File: manuscript/manuscript.Rmd (hunk 14 above)
Comment (cboettig) [reply manuscript/manuscript.Rmd:130]: yeah, kinda begs for citation, though hesitant to call anyone out specifically.  Maybe we can just cite the gym paper on the generic importance of benchmarks?

File: manuscript/manuscript.Rmd (hunk 15 above)
Comment (mlap) [reply manuscript/manuscript.Rmd:141]:
```suggestion
Even the simple models we consider here contain sources of complexity such as hysteresis that are likely commonplace to ecological problems but have not been well represented in RL research to date. 
```

I'm afraid if we don't mention anything it'll come off as vague. Other differentiating sources of complexity I can think of but not necessarily relevant to our environments: huge state-action spaces and poor observability.

Comment (cboettig) [reply manuscript/manuscript.Rmd:141]: good call.  I guess the trouble is the sentence here is about "even the simple models we consider" rather than about ecological problems more generally, so we can't really invoke dimensionality etc. I agree that gym_fishing-v1 doesn't really have any model aspects that aren't considered in classic RL, (probably why most algos do fine on the gym itself after a little training), but I think `conservation-v6` really is distinctly harder for them due to the interactions between being (a) stochastic, (b) highly non-linear, including hysteresis, and (c) being non-autonomous system (changing parameter value over the course of the episode).  Perhaps we can be more precise.  How's this?  

> Our first example using a classic optimal control problem for fisheries turns out to be relatively straight-forward for RL, despite some potential pitfalls we explore. Our second example, though still a simple one-dimensional problem, involves the interaction of stochasticity with non-autonomous dynamics (a slowly changing parameter), and highly non-linear dynamics which produce a tipping point and exhibit hysteresis.  This popular ecological model is often used to examine these very phenomena which are thought to be common in ecological systems but have not been well represented in RL research today. These elements can make this problem considerably harder to solve than models in higher dimensions which assume a deterministic and/or autonomous system.  

(maybe too wordy -- suggestions welcome)

Comment (mlap) [reply manuscript/manuscript.Rmd:141]: Looks good to commit, we could maybe make it without the last sentence but this should be easier to decide after seeing it in the manuscript.

File: manuscript/manuscript.Rmd (hunk 14 above)
Comment (mlap) [reply manuscript/manuscript.Rmd:130]: Hmm thinking more about the citation here. I feel like the animal behavior part might come off as confusing to someone who isn't already familiar with the behavioral rl framing. I think we could drop the behavior part and just cite any algorithm benchmarking paper to cover the "rl agent as sufficiently optimal" assumption here.

**Review by mlap — COMMENTED** [review]
2 more minor comments here.

File: manuscript/manuscript.Rmd (hunk 2 above)
Comment (cboettig) [reply manuscript/manuscript.Rmd:33]: agree, I think the whole sentence can go.

File: manuscript/manuscript.Rmd (hunk 3 above)
Comment (cboettig) [reply manuscript/manuscript.Rmd:115]: ha, I might regret the blunt writing...

File: manuscript/manuscript.Rmd (hunk 10 above)
Comment (cboettig) [reply manuscript/manuscript.Rmd:133]: I agree; line 134 is unlikely to convince anyone who isn't already convinced; so it's really just laboring the point.  I think we just drop that line and leave it with the previous arguments.

File: manuscript/manuscript.Rmd (hunk 14 above)
Comment (cboettig) [reply manuscript/manuscript.Rmd:129]: agree on the idea of a more general citation to benchmarking.  can you drop something in here?

File: manuscript/manuscript.Rmd [hunk 16]
```diff
@@ -112,11 +111,36 @@ Rich, processes-based simulations such as the SORTIE model in forest management
...
 Neural-network-based RL techniques, referred to as *deep RL*, have proven particularly effective in problems involving complex, high-dimensional spaces that have previously proven intractable to classical methods.
 
+While deep RL may have the potential to open up such intractable problems, it also risks making those problems tractable only by stakeholders with access to extensive computational resources and expertise.
+It is notable that the landmark advances cited above have been solved not by academic teams but by specialized research teams of international technology firms such as Alphabet.
+Precise estimates of computational resources used in that research are difficult to establish, but previous estimates benchmarked against commercially avilable cloud computing platforms place the training of a single model at over $30 million [@alphacost],
+and many realistic ecological problems will involve greater complexity than board games of two players on a 19 x 19 grid [@alphago].
```
[the remark this answers is not in the document]
Comment (cboettig) [reply manuscript/manuscript.Rmd:116]: okay I got too snarky which is asking for trouble. revised to state `even greater complexity than these landmark examples [@alphago].`, readers can draw the obvious conclusion without the snark.

**Comment by cboettig** [comment]
@mlap okay, I think I hit everything mentioned above other than adding those citations.  Feel free to merge this when ready.

File: manuscript/manuscript.Rmd [hunk 17]
```diff
@@ -112,11 +110,35 @@ Rich, processes-based simulations such as the SORTIE model in forest management
 Decision-theoretic approaches based on optimal control techniques can only find the best strategy in the simplest of ecological models; the so called "curse of dimensionality" makes problems with a large number of states or actions intractable by conventional methods [@Possingham2006; @Marescot2013; @ferrer-mestres_k-n-momdps_2021; @chades_primer_2021].
```
Comment (karinorman) [inline manuscript/manuscript.Rmd:110]: Maybe assuming previous knowledge of "Decision-theoretic approaches" is appropriate for the intended audience, but might be helpful to have one more sentence of description of the method and/or how it relates to RL. Or just a couple words outlining that this is current best practice approach?

File: manuscript/manuscript.Rmd (hunk 3 above)
Comment (karinorman) [reply manuscript/manuscript.Rmd:114]: To my ear it should be "tractable only _to_ stakeholders"?

File: manuscript/manuscript.Rmd [hunk 18]
```diff
@@ -112,11 +110,35 @@ Rich, processes-based simulations such as the SORTIE model in forest management
...
+It is notable that the landmark advances cited above have been solved not by academic teams but by specialized research teams of international technology firms such as Alphabet.
+Precise estimates of computational resources used in that research are difficult to establish, but previous estimates benchmarked against commercially available cloud computing platforms place the training of a single model at over $30 million [@alphacost],
+and many realistic ecological problems will involve even greater complexity than these landmark examples [@alphago].
+While the history of improved efficiency in computing technology has shown a remarkable ability to reduce such barriers, it has simultaneously moved the leading edge of those capabilities farther beyond reach of traditional ecological research. 
+We believe that ecologists must seek to better understand the design, capabilities and limitations of these algorithms, even though their application will surely require the ambitious collaboration, resources and expertise on par with the scale of the immense environmental and ecological problems we face.
```
Comment (karinorman) [inline manuscript/manuscript.Rmd:118]: This sentence is a little hard for me to follow. Is the message "Ecologists should invest in this even though it will be a huge collaborative, resource-intensive effort"? Bringing it back to "RL requires big things just like our environmental problems require big things" is nice, but I think the message of the sentence gets lost in all the language.

File: manuscript/manuscript.Rmd [hunk 19]
```diff
@@ -112,11 +110,35 @@ Rich, processes-based simulations such as the SORTIE model in forest management
...
+Our implementations of these models have been published to the python-based `PyPi` code archive and include many such variations which represent open problems for RL.
 We include extensive appendices with carefully annotated code which should allow readers to both reproduce and extend this analysis.
 
+We do recognize the need for more complex, "precise" models to derive quantitatively precise management policy in specific management problems such as fisheries, forestry, or climate change mitigation efforts. 
+The examples here should provide ecologists with a greater background to engage in the collaborative development of such methods, while also highlighting challenges which ecological problems pose to existing techniques.
+Even the simple models we consider here contain sources of complexity such as hysteresis that are likely commonplace to ecological problems but have not been well represented in RL research to date. 
+Such work is beyond the scope of any one paper: it will require not only more computational resources, but more collaborations between large teams of ecologists and computer scientists.
```
Comment (karinorman) [inline manuscript/manuscript.Rmd:140]: Not sure what "Such work" refers to, expansion to higher complexity models?
